pkg/registry/apis/query/query.go

```diff
@@ -160,23 +160,29 @@ func (r *queryREST) Connect(connectCtx context.Context, name string, _ runtime.O
 		// Actually run the query
 		rsp, err := b.execute(ctx, req)
 		if err != nil {
-			// log unexpected errors
 			var k8sErr *errorsK8s.StatusError
 			if errors.As(err, &k8sErr) {
-				// we do not need to log 4xx errors as they are expected
 				if k8sErr.ErrStatus.Code >= 500 {
-					b.log.Error("hit unexpected k8s error while executing query", "err", err, "status", k8sErr.Status())
+					if errorsK8s.IsServerTimeout(k8sErr) { // handle context cancelled
+						b.log.Debug("hit a timeout error while executing query, not returning a response to the client as it has already disconnected", "err", err, "status", k8sErr.Status())
+						// No need to send a response to the client, as the client has already disconnected, doing so will cause a subsequent error in the querier
```

Comment:
how sure are we that if `errorsK8s.IsServerTimeout(k8sErr)` then it is 100% the case that the thing that called us, has already disconnected?

i mean, coudn't it be also a case where let's say the apiserver did not respond in the time we allocated for that?

sorry, maybe i'm just missing something obvious here, please help :).

Reply:
ok, have read the related other PR, now it's clearer, sorry. will respond in the other PR.